### DISCUSSION TITLE
Java: Exception in thread "main" java.lang.Error: Unresolved compilation problems

### DISCUSSION THREAD

**Discussion by peter6070** [community/community#52325] [Programming Help]
### Body

Hi! I doing java, and I have a problem.
this is code,
- - - -  - 
class book{
    String title;
    String author;
    public book(){
        this("aaa","bbb");
    }
    public book(String title) {
        this.title = title;
        author = "hhh";
    }
    public book(String t, String a){
        title = t;
        author = a;
    }
}

public class java_6w_1_BookClass {
    public static void main(String[] args) {
        book little = new book("qqq","www");
        book love = new book("eee");
        book them = new book();
        System.out.println(little.title + " " + little.author);
        System.out.println(love.title + " " + love.author);
        System.out.println(them.title + " " + them.author);
    }
}
-- - - - - - - 
and I got the Error message,

Exception in thread "main" java.lang.Error: Unresolved compilation problems: 
        The constructor book(String) is undefined
        The constructor book() is undefined

Why this problem comes up??
#I'm korean, so I used the Translator and my weak english skill. 
Please understand my wrong grammer.
Thank you

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by SMelidoni** [comment]
Hi @peter6070 👋🏽

*English*
The error you're encountering is due to the fact that your `book` class and `java_6w_1_BookClass` class are separate, and you need to nest the `book` class within the `java_6w_1_BookClass`. Here's the corrected code:

*Korean*
당신이 겪고 있는 오류는 `book` 클래스와 `java_6w_1_BookClass` 클래스가 별도로 있기 때문이며, `book` 클래스를 `java_6w_1_BookClass` 내에 중첩시켜야 합니다. 다음은 수정된 코드입니다 (나도 번역기를 사용했어요):

**Code Sample**
```
public class java_6w_1_BookClass {
    static class book {
        String title;
        String author;
        public book() {
            this("aaa", "bbb");
        }
        public book(String title) {
            this.title = title;
            author = "hhh";
        }
        public book(String t, String a) {
            title = t;
            author = a;
        }
    }

    public static void main(String[] args) {
        book little = new book("qqq", "www");
        book love = new book("eee");
        book them = new book();
        System.out.println(little.title + " " + little.author);
        System.out.println(love.title + " " + love.author);
        System.out.println(them.title + " " + them.author);
    }
}
```

Hopefully after making these changes, your code should compile and run without any errors.

이러한 변경 사항을 적용하면 hopefully 코드가 오류 없이 컴파일되고 실행되어야 합니다.

**Reply by peter6070** [reply]
Thank for your comment. Have a great day :)

**Reply by SMelidoni** [reply]
Hi @peter6070, did you manage to find the solution? It would be good if you could post the solution if you found one, or mark the answer that helped you as the accepted solution. That way other users can find the solution a lot easier! 

안녕하세요 @peter6070님, 해결책을 찾으셨나요? 만약 해결책을 찾으셨다면, 그 해결책을 게시해 주시거나 도움이 된 답변을 수락된 해결책으로 표시해 주시면 좋겠습니다. 그렇게 하면 다른 사용자들이 해결책을 훨씬 쉽게 찾을 수 있습니다!

**Comment by sergeyleschev** [comment]
It looks like you are missing constructors in your book class. Try adding a constructor with no arguments and another one with a String argument to fix the issue.

**Reply by peter6070** [reply]
Thank for your comment. Have a great day :)
